lib/api/apiUtils/object/versioning.js

```diff
@@ -170,7 +170,7 @@ function processVersioningState(mst, vstat) {
         options.versioning = true;
         if (mst.exists) {
             // store master version in a new key
-            const versionId = mst.isNull ? mst.versionId : nonVersionedObjId;
+            const versionId = (mst.isNull && mst.versionId) ? mst.versionId : nonVersionedObjId;
```

Comment:
Is this meant to support the existing "wrong" master with `isNull` but no `versionId`, that we want to avoid creating from now on? I'm fine with it, just wondering if that's the purpose.

Reply:
> In certain cases, a master version may not have a versionId and be set as null (isNull:true)

Actually if I understand this, it is a "normal" situation so we need to take care of it properly.

(It's much likely a bug that we should fix by the way, we should not set the `isNull` flag in that case, but obviously we need to support it due to the existing data).

Reply:
> Is this meant to support the existing "wrong" master with isNull but no versionId, that we want to avoid creating from now on? I'm fine with it, just wondering if that's the purpose.

Yes, it is to support this case created by putMetadata.

> Actually if I understand this, it is a "normal" situation so we need to take care of it properly.

Correct.

Also, as you said, we might want to fix it. I will create a ticket for it.

Reply:
[CLDSRV-509](https://scality.atlassian.net/browse/CLDSRV-509)